canonical-web-and-design/maas-ui
# Set up KVM details components.

## Done

- Created bare-bones components for  `KVMConfiguration` and `KVMSummary`.
- Added `KVMDetailsHeader` component with action dropdown
- Moved `KVMActionFormWrapper` from KVM list to kvm/components and used in KVMDetailsHeader so the refresh and delete forms can be used here too
- Updated refresh and delete forms to work in KVMDetails

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/kvm and click on one of the KVMs
- Check that the KVM name and composed machine count is correct
- Check that you can navigate between summary and configuration
- Check that you can open the delete and refresh forms from the action dropdown ("Compose" is currently empty, to be built in another issue)
- Check that you can refresh a KVM
- Check that you can delete a KVM and be redirected to the KVM list

## Fixes

Fixes #1366
